newsletter/newsletter_queries.py

```diff
@@ -0,0 +1,202 @@
+"""Queries for summary report of the weeklly newsletter"""
+from psycopg2 import connect
+from os import environ as ENV
+from dotenv import load_dotenv
+from psycopg2.extras import RealDictCursor
+import pandas as pd
+
+
+def get_db_connection():
+    """Connect to the postgres database managed by RDS."""
+    load_dotenv()
+
+    return connect(database=ENV["DB_NAME"],
+                   user=ENV["DB_USERNAME"],
+                   password=ENV["DB_PASSWORD"],
+                   host=ENV["DB_IP"],
+                   port=ENV["DB_PORT"])
+
+
+def get_weekly_average(conn) -> int:
+    """total demand, average demand, average price for the week"""
+    with conn.cursor() as cur:
+        cur.execute("""
+            SELECT SUM(demand)AS total_demand, AVG(demand)AS average_demand, 
+                          AVG(price)AS average_price
+            FROM power_reading
+            WHERE date_time >= NOW() - INTERVAL '7 days';
+                    """)
+        total = cur.fetchone()
+
+    return [round(t, 2) for t in total]
+
+
+def weekly_highest_price(conn) -> float:
+    """highest price for the week"""
+    with conn.cursor() as cur:
+        cur.execute("""
+            SELECT price, date_time
+            FROM power_reading
+            WHERE date_time >= NOW() - INTERVAL '7 days'
+            ORDER BY price DESC, date_time ASC
+            LIMIT 1;
+        """)
+        price = cur.fetchone()
+    return round(price[0], 2)
```

Comment:
```suggestion
            SELECT price, date_time
            FROM power_reading
            WHERE date_time >= NOW() - INTERVAL '7 days'
            AND price = (SELECT MAX(price) FROM power_reading);
            
            price = cur.fetchall()
           
      return price
```

This is more robust as it will return all of the date/times that price was reached. This is important in the case where the same max price has been reached multiple times in a week. And there is no need to round since the values in the table are already rounded.